pytket/extensions/qiskit/tket_backend.py

```diff
@@ -12,36 +12,39 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import inspect
 from typing import Optional, Any
+from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping  # type: ignore
 from qiskit.circuit.quantumcircuit import QuantumCircuit  # type: ignore
-from qiskit.providers.backend import BackendV1  # type: ignore
-from qiskit.providers.models import QasmBackendConfiguration  # type: ignore
+from qiskit.providers.backend import BackendV2  # type: ignore
 from qiskit.providers import Options  # type: ignore
+from qiskit.transpiler import CouplingMap, Target  # type: ignore
 from pytket.extensions.qiskit import AerStateBackend, AerUnitaryBackend
 from pytket.extensions.qiskit.qiskit_convert import qiskit_to_tk, _gate_str_2_optype_rev
 from pytket.extensions.qiskit.tket_job import TketJob, JobInfo
 from pytket.backends import Backend
 from pytket.passes import BasePass
-from pytket.predicates import (
-    NoClassicalControlPredicate,
-    GateSetPredicate,
-    CompilationUnit,
-)
+from pytket.predicates import GateSetPredicate, CompilationUnit
 from pytket.architecture import FullyConnected
 
 
 def _extract_basis_gates(backend: Backend) -> list[str]:
+    standard_gate_mapping = get_standard_gate_name_mapping()
     for pred in backend.required_predicates:
         if type(pred) == GateSetPredicate:
-            return [
-                _gate_str_2_optype_rev[optype]
-                for optype in pred.gate_set
-                if optype in _gate_str_2_optype_rev.keys()
-            ]
+            basis_gates = []
+            for optype in pred.gate_set:
+                if optype in _gate_str_2_optype_rev.keys():
+                    gate_name = _gate_str_2_optype_rev[optype]
+                    if gate_name in standard_gate_mapping:
+                        gate_obj = standard_gate_mapping[gate_name]
+                        if gate_obj.num_qubits in [1, 2] or inspect.isclass(gate_obj):
```

Comment:
See https://github.com/Qiskit/qiskit/blob/2ef371ae0d159a6dfd643805f3e5e5fdec37ab88/qiskit/transpiler/target.py#L1066 -- have to restrict the gateset to match what is accepted by the `Target` method.

Reply:
Do I understand correctly that the `gate_obj` variable is not always a class? Presumably thats why the inspect check is needed.

Reply:
Yes, some gates such as CCX were rejected when I omitted this check. I am not sure when `gate_obj` is a class, but simply copied that from the code linked above.